pkg/platform/kube/platform.go

```diff
@@ -1260,3 +1272,47 @@ func (p *Platform) getDefaultServiceType() (v1.ServiceType, error) {
 		return "", errors.New("Not a valid configuration instance")
 	}
 }
+
+func (p *Platform) validateAPIGatewayFunctionsHaveNoIngresses(platformAPIGateway *platform.APIGatewayConfig) error {
+
+	// check ingresses on every upstream function
+	for _, upstream := range platformAPIGateway.Spec.Upstreams {
```

Comment:
use err groups, make it run in parallel.
this validation can be very expensive and might stall create function by X amount of upstream functions